### PR TITLE
add user-system-information funktionality into api-integration

### PR DESCRIPTION
**Pull request by hiddenmind3**

### CODE REVIEW

**Comment by hiddenmind3** [comment]
forgot to rebase user_system_information. Closing PR

File: src/main/java/net/adoptium/IndexResource.java [hunk 1]
```diff
@@ -7,10 +7,7 @@
 import org.eclipse.microprofile.rest.client.inject.RestClient;
 
 import javax.inject.Inject;
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.*;
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/IndexResource.java:10]: Let's try to not use * imports (can be defined in IDE)

File: src/main/java/net/adoptium/UserAgentParser.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        defaultArchOfOS.put(OperatingSystem.linux, Architecture.x64);
+        defaultArchOfOS.put(OperatingSystem.windows, Architecture.x64);
+        defaultArchOfOS.put(OperatingSystem.mac, Architecture.x64);
+        defaultArchOfOS.put(OperatingSystem.solaris, Architecture.x64);
+        defaultArchOfOS.put(OperatingSystem.aix, Architecture.ppc64);
+        defaultArchOfOS.put(OperatingSystem.valueOf("alpine-linux"), Architecture.x64);
+    }
+
+    public static UserSystem getOsAndArch(String userAgent) {
+        userAgent = userAgent.toLowerCase(Locale.ENGLISH);
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/UserAgentParser.java:50]: try to define method params as final. Mutating a method parameter is not best practice. Better create a new (final) variable for this.

File: src/main/java/net/adoptium/UserAgentParser.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.Locale;
+import java.util.Map;
+
+public class UserAgentParser {
+
+    private static final Map<OperatingSystem, String[]> osMap;
+
+    static {
+        osMap = new LinkedHashMap<>();
+        osMap.put(OperatingSystem.linux, new String[]{"linux"});
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/UserAgentParser.java:16]: All 3 maps looks like a great use case for enums :)

File: src/main/java/net/adoptium/UserSystem.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+public class UserSystem {
+    private OperatingSystem os;
+    private Architecture arch;
+
+    public void UserSystem(OperatingSystem os, Architecture arch) {
+        this.os = os;
+        this.arch = arch;
+    }
+
+    public void setOs(OperatingSystem os) {
```
Comment (hendrikebbers) [inline src/main/java/net/adoptium/UserSystem.java:15]: Does it really makes sense to change OS/Arch after object creation? Maybe it would be better to define the fields as final and remove the setter

File: src/main/resources/templates/index.html [hunk 5]
```diff
@@ -5,36 +5,8 @@
...
-                <h3>Prebuilt OpenJDK</h3>
-                <p>Just the best.</p>
-                <p><code>v1.2</code></p>
-                <p><a href="" class="guide-link" target="_blank">Related guide section...</a></p>
-            </div>
-        </div>
+        <p>User-Agent: {userAgent}<br>Os: {os}<br>Architecture: {arch}</p>
     </div>
     <style type="text/css" scoped>
         h1, h2, h3, h4, h5, h6 {
```
Comment (hendrikebbers) [inline src/main/resources/templates/index.html:12]: Let's directly externalise the CSS

File: src/test/java/net/adoptium/UserAgentParserTest.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+class UserAgentParserTest {
+
+    @BeforeEach
+    void setUp() {
+
+    }
+
+    @Test
+    void getOsAndArchExpectedLinuxAsDefault() {
```
Comment (hendrikebbers) [inline src/test/java/net/adoptium/UserAgentParserTest.java:18]: Such tests are always way better readable when you follow and mark the simple "given-when-than" pattern:

```
//given
final String userAgent = "Mozilla/5.0 (X11; Linux x86_64; rv:2.1.1) Gecko/ Firefox/5.0.1";

//when
final UserSystem osArch = UserAgentParser.getOsAndArch(userAgent);

//than
assertEquals(osArch.getArch(), Architecture.x64);
assertEquals(osArch.getOs(), OperatingSystem.linux);
```

File: src/test/java/net/adoptium/UserAgentParserTest.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        String userAgent = "Mozilla/5.0 (X11; Linux x86_64; rv:2.1.1) Gecko/ Firefox/5.0.1";
+        UserSystem osArch = UserAgentParser.getOsAndArch(userAgent);
+        assertEquals(osArch.getArch(), Architecture.x64);
```
Comment (hendrikebbers) [inline src/test/java/net/adoptium/UserAgentParserTest.java:21]: first param is the expected one ;)

File: src/test/java/net/adoptium/UserAgentParserTest.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    @Test
+    void getOsAndArchExpectedWindowsX32Third() {
+        String userAgent = "Mozilla/5.0 (Windows NT 10.0; x86_32; rv:87.0) Gecko/20100101 Firefox/87.0";
+        UserSystem osArch = UserAgentParser.getOsAndArch(userAgent);
+        assertEquals(osArch.getArch(), Architecture.x32);
+        assertEquals(osArch.getOs(), OperatingSystem.windows);
+    }
+
+    @Test
+    void getOsAndArchTestLinuxNotWindows() {
```
Comment (hendrikebbers) [inline src/test/java/net/adoptium/UserAgentParserTest.java:110]: All your tests only take care of the lucky path. Add some tests that have a user agent that just do not work or even `null` as user agent.
